src/fsharp/fsc.fs

```diff
@@ -1528,149 +1528,112 @@ module StaticLinker =
                         | Some provAssemStaticLinkInfo -> yield (importedBinary,provAssemStaticLinkInfo) ]
 #endif
         if tcConfig.compilingFslib && tcConfig.compilingFslib20.IsSome then 
-            (fun ilxMainModule -> LegacyFindAndAddMscorlibTypesForStaticLinkingIntoFSharpCoreLibraryForNet20 (tcConfig, ilGlobals, ilxMainModule))
+            [], (fun ilxMainModule -> LegacyFindAndAddMscorlibTypesForStaticLinkingIntoFSharpCoreLibraryForNet20 (tcConfig, ilGlobals, ilxMainModule))
           
         elif not tcConfig.standalone && tcConfig.extraStaticLinkRoots.IsEmpty 
 #if EXTENSIONTYPING
              && providerGeneratedAssemblies.IsEmpty 
 #endif
              then 
-            (fun ilxMainModule -> ilxMainModule)
+            [], (fun ilxMainModule -> ilxMainModule)
         else 
-            (fun ilxMainModule  ->
```

Comment:
@dsyme Just to clarify a little more: Previously, the below code that calculates the `generatedILTypeDefs` lived inside this returned static linker function ... The problem was that this function was executed _after_ the compilation, so those `ILTypeDefs` were not available to be extended. None of the following code has changed, but it has been moved outside that function, and now a tuple of the `generatedILTypeDefs` and the static linker function are returned.